utils/hct/gen_intrin_main.txt

```diff
@@ -380,6 +380,14 @@ void [[]] Barrier(in NodeRecordOrUAV o, in uint SemanticFlags);
 
 uint [[]] GetRemainingRecursionLevels();
 
+void [[]] __builtin_MatVecMul(out numeric<c> output_vector, in bool isOutputSigned, in numeric<c2> input_vector, in bool isInputUnsigned, in uint input_interpretation, in ByteAddressBuffer matrix_buffer, in uint matrix_offset, in uint matrix_interpretation, in uint M, in uint K, in uint matrix_layout, in bool matrix_is_transposed, in uint matrix_stride);
+ 
+void [[]] __builtin_MatVecMulAdd(out numeric<c> output_vector, in bool isOutputUnsigned, in numeric<c2> input_vector, in bool isInputUnsigned, in uint input_interpretation, in ByteAddressBuffer matrix_buffer, in uint matrix_offset, in uint matrix_interpretation, in uint M, in uint K, in uint matrix_layout, in bool matrix_is_transposed, in uint matrix_stride, in ByteAddressBuffer bias_vector, in uint bias_offset, in uint bias_interpretation);
+ 
+void [[]] __builtin_OuterProductAccumulate(in numeric<c> input_vector1, in numeric<c2> input_vector2, in RWByteAddressBuffer matrix_buffer, in uint matrix_offset, in uint matrix_interpretation, in uint matrix_layout, in uint matrix_stride);
+ 
+void [[]] __builtin_VectorAccumulate(in numeric<c> input_vector, in RWByteAddressBuffer matrix_buffer, in uint matrix_offset);
+
```

Comment:
These aren't actually in the DX namespace.

See `MaybeReorderThread` (I think) for an example of how to do this.  This also has the `min_sm` stuff that would be needed to prevent these from showing up in older shader models.

Reply:
Oh right! That wasn't checked in yet when I made these changes. Looks like it is there now.

Reply:
See https://github.com/microsoft/DirectXShaderCompiler/issues/7301 - we'll leave this as it is for the preview and will address the namespace for the new intrinsics later.